packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/usePaletteReferencePosition.tsx

```diff
@@ -10,13 +10,16 @@
  * Contributors:
  *     Obeo - initial API and implementation
  *******************************************************************************/
-import { useViewport } from 'reactflow';
-import { UsePaletteReferencePositionValue } from './usePaletteReferencePosition.types';
+import { useStoreApi } from 'reactflow';
 import { useDiagramElementPalette } from './useDiagramElementPalette';
 import { useDiagramPalette } from './useDiagramPalette';
+import { UsePaletteReferencePositionValue } from './usePaletteReferencePosition.types';
 
 export const usePaletteReferencePosition = (): UsePaletteReferencePositionValue => {
-  const { x: viewportX, y: viewportY, zoom: viewportZoom } = useViewport();
+  const store = useStoreApi();
+  const { transform } = store.getState();
+  const [viewportX, viewportY, viewportZoom] = [...transform];
```

Comment:
What is the point of this change?

Reply:
When using useViewport() hook, the component will re render each time the viewport change. 
With this change we avoid unnecessary rendering.